### PR TITLE
OAK-10792 - Rename DetailedGC to FullGC

### PR DESCRIPTION
**Pull request by daniancu**
Switching to FullGC instead of DetailedGC everywhere, method names, constants, arguments etc

### CODE REVIEW

**Comment by reschke** [comment]
Can you please open a JIRA ticket to track this change?

File: oak-run/src/main/java/org/apache/jackrabbit/oak/run/RevisionsCommand.java [hunk 1]
```diff
@@ -46,19 +31,32 @@
 import org.apache.jackrabbit.oak.plugins.document.NodeDocument;
 import org.apache.jackrabbit.oak.plugins.document.RevisionContextWrapper;
 import org.apache.jackrabbit.oak.plugins.document.SweepHelper;
+import org.apache.jackrabbit.oak.plugins.document.VersionGCOptions;
```
Comment (reschke) [inline oak-run/src/main/java/org/apache/jackrabbit/oak/run/RevisionsCommand.java]: Please do not move imports around (unless the change is exactly for thar purpose)

File: oak-run/src/main/java/org/apache/jackrabbit/oak/run/RevisionsCommand.java [hunk 2]
```diff
@@ -92,9 +90,9 @@ public class RevisionsCommand implements Command {
             "  collect        perform garbage collection",
             "  reset          clear all persisted metadata",
             "  sweep          clean up uncommitted changes",
-            "  detailedGC     perform detailed garbage collection i.e. remove unmerged branch commits, old ",
+            "  fullGC         perform detailed garbage collection i.e. remove unmerged branch commits, old ",
```
Comment (rishabhdaim) [inline oak-run/src/main/java/org/apache/jackrabbit/oak/run/RevisionsCommand.java]:
```suggestion
            "  fullGC         perform full garbage collection i.e. remove unmerged branch commits, old ",
```

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java [hunk 3]
```diff
@@ -311,10 +311,10 @@
     @AttributeDefinition(
             name = "Document Node Store Detailed GC",
             description = "Boolean value indicating whether Detailed GC should be enabled for " +
```
Comment (rishabhdaim) [inline oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java]:
```suggestion
            description = "Boolean value indicating whether Full GC should be enabled for " +
```

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java [hunk 4]
```diff
@@ -311,10 +311,10 @@
     @AttributeDefinition(
             name = "Document Node Store Detailed GC",
```
Comment (rishabhdaim) [inline oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java]:
```suggestion
            name = "Document Node Store Full GC",
```

File: oak-run/src/main/java/org/apache/jackrabbit/oak/run/RevisionsCommand.java (hunk 1 above)
Comment (daniancu) [reply oak-run/src/main/java/org/apache/jackrabbit/oak/run/RevisionsCommand.java]: sorry for that, I've set them back

**Comment by daniancu** [comment]
> Can you please open a JIRA ticket to track this change?

https://issues.apache.org/jira/browse/OAK-10792

**Comment by daniancu** [comment]
@reschke @rishabhdaim can you take an other look?

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java [hunk 5]
```diff
@@ -311,10 +311,10 @@
...
-                    "document node store or not. The Default value is " + DEFAULT_DETAILED_GC_ENABLED +
+                    "document node store or not. The Default value is " + DEFAULT_FULL_GC_ENABLED +
                     ". Note that this value can be overridden via framework " +
-                    "property 'oak.documentstore.detailedGCEnabled'")
-    boolean detailedGCEnabled() default DEFAULT_DETAILED_GC_ENABLED;
+                    "property 'oak.documentstore.fullGCEnabled'")
+    boolean fullGCEnabled() default DEFAULT_FULL_GC_ENABLED;
 
     @AttributeDefinition(
             name = "Document Node Store Embedded Verification for Detailed GC",
```
Comment (rishabhdaim) [inline oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java]:
```suggestion
            name = "Document Node Store Embedded Verification for Full GC",
```

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java [hunk 6]
```diff
@@ -748,23 +748,23 @@ private VersionGCStats gc(long maxRevisionAgeInMillis) throws IOException {
...
-                        phases.stats.ignoredDetailedGCDueToCheckPoint = true;
+                if (fullGCEnabled) {
+                    stats.fullGCActive.start();
+                    if (rec.ignoreFullGCDueToCheckPoint) {
+                        phases.stats.ignoredFullGCDueToCheckPoint = true;
                         monitor.skipped("Checkpoint prevented detailed revision garbage collection");
                     } else {
                         final RevisionVector headRevision = nodeStore.getHeadRevision();
-                        monitor.info("Looking at revisions in {} for detailed GC", rec.scopeDetailedGC);
+                        monitor.info("Looking at revisions in {} for detailed GC", rec.scopeFullGC);
```
Comment (rishabhdaim) [inline oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java]:
```suggestion
                        monitor.info("Looking at revisions in {} for full GC", rec.scopeFullGC);
```

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java [hunk 7]
```diff
@@ -850,9 +850,9 @@ protected boolean removeEldestEntry(Entry<Path, Boolean> eldest) {
...
                                 lastDoc = doc;
                                 // collect the data to delete in next step
-                                if (phases.start(GCPhase.DETAILED_GC_COLLECT_GARBAGE)) {
+                                if (phases.start(GCPhase.FULL_GC_COLLECT_GARBAGE)) {
                                     gc.collectGarbage(doc, phases);
-                                    phases.stop(GCPhase.DETAILED_GC_COLLECT_GARBAGE);
+                                    phases.stop(GCPhase.FULL_GC_COLLECT_GARBAGE);
                                 }
 
                                 final Long modified = lastDoc.getModified();
```
Comment (rishabhdaim) [inline oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java:858]: Update below lines: 860 & 862

`collectDetailedGarbage` --> `collectFullGarbage`

File: oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java [hunk 8]
```diff
@@ -18,6 +18,46 @@
...
+import org.jetbrains.annotations.NotNull;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
```
Comment (rishabhdaim) [inline oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java]: could you please revert the import statement changes?

File: oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java [hunk 9]
```diff
@@ -1045,7 +1044,7 @@ public void testGCDeletedNonBundledProps() throws Exception {
         store1.merge(b1, EmptyHook.INSTANCE, CommitInfo.EMPTY);
 
         // enable the detailed gc flag
```
Comment (rishabhdaim) [inline oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java]:
```suggestion
        // enable the Full gc flag
```

Comment (rishabhdaim) [reply oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java]: Same for all the comment with `// enable the detailed gc flag`

File: oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java [hunk 10]
```diff
@@ -2243,8 +2242,8 @@ public void testDeletedPropsAndUnmergedBCWithCollisionWithDryRunMode() throws Ex
         store1.runBackgroundOperations();
 
         // enable the detailed gc flag
```
Comment (rishabhdaim) [inline oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java]:
```suggestion
        // enable the Full gc flag
```

**Review by rishabhdaim — CHANGES_REQUESTED** [review]

**Comment by rishabhdaim** [comment]
We should only merge this PR after incorporating the integration branch changes into the trunk.
cc @stefan-egli @reschke

File: oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java (hunk 9 above)
Comment (daniancu) [reply oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java]: @rishabhdaim I think I fixed them all, can you review pls?

File: oak-run/src/main/java/org/apache/jackrabbit/oak/run/RevisionsCommand.java (hunk 2 above)
Comment (daniancu) [reply oak-run/src/main/java/org/apache/jackrabbit/oak/run/RevisionsCommand.java]: fixed

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java (hunk 3 above)
Comment (daniancu) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java]: fixed

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java (hunk 4 above)
Comment (daniancu) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java]: fixed

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java (hunk 5 above)
Comment (daniancu) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java]: fixed

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java (hunk 6 above)
Comment (daniancu) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java]: fixed

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java (hunk 7 above)
Comment (daniancu) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java:858]: fixed

File: oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java (hunk 8 above)
Comment (daniancu) [reply oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java]: fixed

File: oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java (hunk 10 above)
Comment (daniancu) [reply oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java]: fixed

**Comment by stefan-egli** [comment]
> We should only merge this PR after incorporating the integration branch changes into the trunk.

@rishabhdaim does it otherwise cause merge conflicts? I understand the concern - on the other hand this should only change new code, so wouldn't expect additional merge conflicts? (but if it would, then yes)

**Review by rishabhdaim — APPROVED** [review]

**Comment by stefan-egli** [comment]
@daniancu , FTR, we should prefix the commit message with the jira ticket (agreed upon practice, helps navigate commits)

**Pull request merged**